image/tarexport/load.go

```diff
@@ -422,8 +423,8 @@ func checkCompatibleOS(imageOS string) error {
 		return fmt.Errorf("cannot load %s image on %s", imageOS, runtime.GOOS)
 	}
 	// Finally, check the image OS is supported for the platform.
-	if err := system.ValidatePlatform(system.ParsePlatform(imageOS)); err != nil {
-		return fmt.Errorf("cannot load %s image on %s: %s", imageOS, runtime.GOOS, err)
```

Comment:
Do we still need the check for incompatible platforms? Looks like `platforms.Parse()` does not check for that

Reply:
I think we do need a method to make sure the server can get a valid {OS,Arch} pair, but currently we don't have yet because `Arch` is not supported(take look at the `ValidatePlatform()` code). I prefer to add that method into `containerd` code just like the `Parse()` does in the future. 
As the first step, this PR will keep the original functionalities unchanged but begin to support OCI spec `Platform` instead of current temp `OS` "platform" handling. Actually now we don't have a multi-platform capable daemon server, so even we specify the `--platform` option, it doesn't work meaning this will not affect the final image generated. We can reach that target gradually IMO, beginning from the client side, then to the server.